### ISSUE TITLE
ob run fails on Linux

### ISSUE THREAD

**Issue by StevenXL** [obsidiansystems/obelisk#910]
Hi folks,

I am running `ob run` on Linux; the command doesn't seem to output any explicit error:

```
[steven@ip-172-31-44-251 ob-test]$ ob run
./.obelisk/impl: command not cached, building ...
DONE Built on ./.obelisk/impl [command]
```

However, when I try to access localhost:8000, there is no application listening on that port. I have confirmed that `.confing/common/route` contains `http://localhost:8000`. 

When I stop the process, the following is outputted:

```
Process exited with code 1; /nix/store/4z4336r8yhyznz3fiscvz53wgdj43kjb-nix-2.3.11/bin/nix-shell -E $'{root, pkgs, shell}: ((import root {}).passthru.__unstable__.self.extend (_: _: {shellPackages = builtins.fromJSON pkgs;})).project.shells.${shell}' --arg root ./. --argstr pkgs $'{"backend":"/home/steven/ob-test/backend","common":"/home/steven/ob-test/common","frontend":"/home/steven/ob-test/frontend","obelisk-generated-static":"/nix/store/0r34z6nywaikj6dmsdv856v0d5ylar1m-asset-manifest-haskellManifest"}' --argstr shell ghc --run $'export $\'NIX_PATH=nixpkgs=/nix/store/yxa12wh8fc6j8w61y3jn27vw8dhydby7-source\' ; bash -c \'type -p ghc\''
```

`nix-env --version` => `nix-env (Nix) 2.7.0`

`cat /etc/os-release` => 

```
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
```

**Comment by alexfmpe** [comment]
One possibility is that the first run will build a lot of dependencies first. The verbose flag might give more insight: 
`ob run -v`

**Comment by runeksvendsen** [comment]
Hi @StevenXL, thank you for the report.

> When I stop the process, the following is outputted:
> 
> ```
> Process exited with code 1; /nix/store/4z4336r8yhyznz3fiscvz53wgdj43kjb-nix-2.3.11/bin/nix-shell -E $'{root, pkgs, shell}: ((import root {}).passthru.__unstable__.self.extend (_: _: {shellPackages = builtins.fromJSON pkgs;})).project.shells.${shell}' --arg root ./. --argstr pkgs $'{"backend":"/home/steven/ob-test/backend","common":"/home/steven/ob-test/common","frontend":"/home/steven/ob-test/frontend","obelisk-generated-static":"/nix/store/0r34z6nywaikj6dmsdv856v0d5ylar1m-asset-manifest-haskellManifest"}' --argstr shell ghc --run $'export $\'NIX_PATH=nixpkgs=/nix/store/yxa12wh8fc6j8w61y3jn27vw8dhydby7-source\' ; bash -c \'type -p ghc\''
> ```

Please run this command manually and paste the output here. Ie. the command:

```
/nix/store/4z4336r8yhyznz3fiscvz53wgdj43kjb-nix-2.3.11/bin/nix-shell -E $'{root, pkgs, shell}: ((import root {}).passthru.__unstable__.self.extend (_: _: {shellPackages = builtins.fromJSON pkgs;})).project.shells.${shell}' --arg root ./. --argstr pkgs $'{"backend":"/home/steven/ob-test/backend","common":"/home/steven/ob-test/common","frontend":"/home/steven/ob-test/frontend","obelisk-generated-static":"/nix/store/0r34z6nywaikj6dmsdv856v0d5ylar1m-asset-manifest-haskellManifest"}' --argstr shell ghc --run $'export $\'NIX_PATH=nixpkgs=/nix/store/yxa12wh8fc6j8w61y3jn27vw8dhydby7-source\' ; bash -c \'type -p ghc\''
```

**Comment by jrpotter** [comment]
I ended up seeing this same issue on a fedora machine. Running `ob run -v` (after setting `LANG=C.UTF-8` as mentioned in https://github.com/obsidiansystems/obelisk/issues/908) showed nix was taking a while to get the ghc-8.6.5 package, though it looked like the project had hanged. I gave it another 20 minutes and it eventually continued.

It would be nice to receive some output from obelisk indicating it is still doing something, even if just waiting on one package.
